### PR TITLE
Add A3 ultra blueprint to deploy VMs and update documentation

### PR DESCRIPTION
**Pull request by harshthakkar01**
Co-authored-by: carsondunbar@[google.com](http://google.com/)

### Submission Checklist

NOTE: Community submissions can take up to 2 weeks to be reviewed.

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cluster Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+      zone: $(vars.zone)
+
+  google-beta:
+    source: hashicorp/google-beta
+    version: 6.13.0
+    configuration:
+      project: $(vars.project_id)
+      region: $(vars.region)
+      zone: $(vars.zone)
+
```
Comment (tpdownes) [inline examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml:48]:
```suggestion
```

File: examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      local_mount: /home
+      reserved_ip_range: $(vars.filestore_ip_range)
+    outputs:
+    - network_storage
+
+  - id: startup-script
+    source: modules/scripts/startup-script
+    settings:
+      configure_ssh_host_patterns:
+      - $(vars.deployment_name)-beowulf-*
```
Comment (tpdownes) [inline examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml:121]: please create

```
vars:
  hostname_prefix: $(vars.deployment_name)-beowulf
```

and then use that variable in the 2 locations

File: examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            ipv6_access_config=[],
+            alias_ip_range=[]
+          }],
+          a3ultra-rdma-net.subnetwork_interfaces,
+        ))
+
+  - id: wait-for-vms
+    source: community/modules/scripts/wait-for-startup
+    settings:
+      instance_name: $(a3ultra-vms.name[0])
```
Comment (tpdownes) [inline examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml:166]:
```suggestion
      instance_names: $(a3ultra-vms.name)
```

**Review by tpdownes — CHANGES_REQUESTED** [review]

File: examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml (hunk 1 above)
Comment (harshthakkar01) [reply examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml:49]: Trying to understand, what is the suggestion here ?

File: examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+      local_mount: /home
+      reserved_ip_range: $(vars.filestore_ip_range)
+    outputs:
+    - network_storage
+
+  - id: startup-script
+    source: modules/scripts/startup-script
+    settings:
+      configure_ssh_host_patterns:
+      - $(hostname_prefix)-*
```
Comment (tpdownes) [inline examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml:122]:
```suggestion
      - $(vars.hostname_prefix)-*
```

File: examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+  - id: a3ultra-vms
+    source: modules/compute/vm-instance
+    use: [startup-script, homefs]
+    settings:
+      machine_type: a3-ultragpu-8g
+      instance_count: 2
+      name_prefix: $(hostname_prefix)
```
Comment (tpdownes) [inline examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml:130]:
```suggestion
      name_prefix: $(vars.hostname_prefix)
```

**Review by tpdownes — CHANGES_REQUESTED** [review]

File: examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml (hunk 1 above)
Comment (tpdownes) [reply examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml:49]: Removing the unnecessary blocks (6.13.0 is the default)

Comment (harshthakkar01) [reply examples/machine-learning/a3-ultragpu-8g/a3ultra-vm.yaml]: gotcha. removed.

**Review by tpdownes — APPROVED** [review]

**Pull request merged**